io/src/main/java/org/commonjava/maven/ext/io/rest/DefaultTranslator.java

```diff
@@ -108,14 +110,29 @@
      * @param restMinSize minimum size for the call
      * @param repositoryGroup the group to pass to the endpoint.
      * @param incrementalSerialSuffix the suffix to pass to the endpoint.
+     * @param headers the headers to pass to the endpoint
      */
-    public DefaultTranslator( String endpointUrl, int restMaxSize, int restMinSize, String repositoryGroup, String incrementalSerialSuffix )
+    public DefaultTranslator( String endpointUrl, int restMaxSize, int restMinSize, String repositoryGroup, String incrementalSerialSuffix, Map<String, String> headers )
     {
         this.repositoryGroup = repositoryGroup;
         this.incrementalSerialSuffix = incrementalSerialSuffix;
         this.endpointUrl = endpointUrl + ( isNotBlank( endpointUrl ) ? endpointUrl.endsWith( "/" ) ? "" : "/" : "");
         this.initialRestMaxSize = restMaxSize;
         this.initialRestMinSize = restMinSize;
+        this.headers = headers;
+    }
+
+    /**
+     * @param endpointUrl is the URL to talk to.
+     * @param restMaxSize initial (maximum) size of the rest call; if zero will send everything.
+     * @param restMinSize minimum size for the call
+     * @param repositoryGroup the group to pass to the endpoint.
+     * @param incrementalSerialSuffix the suffix to pass to the endpoint.
+     */
+    public DefaultTranslator( String endpointUrl, int restMaxSize, int restMinSize, String repositoryGroup, String incrementalSerialSuffix )
+    {
+        this ( endpointUrl, restMaxSize, restMinSize, repositoryGroup, incrementalSerialSuffix,
+               Collections.emptyMap() );
```

Comment:
Why do we need another ctor?

Reply:
> Why do we need another ctor?

It was used to pass the headers through to this class, but shouldn't be needed if using properties.

Reply:
Ok!

Reply:
> Ok!

I moved the code to `RESTState`, but it still needs to pass the `restHeaders` like all these other arguments here are (which also come from user properties). Is the issue that it's adding an argument, or just that there should be a single constructor and not two? Most of the calls wouldn't pass any headers, just the call from the `RESTState` does.